### ISSUE TITLE
[a11y]: Dropdown has critical violation when the dropdown list is open: None of the descendent elements with "option" role is tabbable

### ISSUE THREAD

**Issue by cvikram21** [carbon-design-system/carbon#13110]
### Package

carbon-components, carbon-components-react

### Browser

Firefox

### Operating System

MacOS

### Package version

7.57.4

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker: Latest Deployment

### Description

When the dropdown list is open, IBM Equal Access Accessibility Checker gives a violation as shown in the attached screenshot. This happens in both V10 and V11 versions of carbon components.

![image](https://user-images.githubusercontent.com/22382586/217274512-cff5843b-fd34-4e38-8e99-bcba85232d1a.png)
![image](https://user-images.githubusercontent.com/22382586/217274577-3faa4b80-c994-4c6e-a14a-8cf82dccfb9d.png)

### Reproduction/example

https://carbondesignsystem.com/components/dropdown/usage/

### Steps to reproduce

1. Go to the carbon design system website.
2. Select the component Dropdown.
3. Open the dropdown and run the Accessibility checker tool. The violation will be reported by the tool

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Yes this is a violation but despite the options not being tabbable, they are still operable via arrow keys and are properly announced by screenreaders.

Based on a [conversation](https://ibm-studios.slack.com/archives/G01GCBCGTPV/p1675865500656539) with IBMa, Dropdown needs to be revised to be a [Select-Only Combobox](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).

This means the Dropdown would be `role=combobox`. If the listbox was pointed to by a combobox, this violation would no longer appear.
